platform/src/main/java/org/stellar/anchor/platform/CallbackApiBeans.java

```diff
@@ -49,29 +33,66 @@ OkHttpClient httpClient() {
 
   @Bean
   UniqueAddressIntegration uniqueAddressIntegration(
-      Sep31Config sep31Config, OkHttpClient httpClient, AuthHelper authHelper, Gson gson) {
+      AppConfig appConfig,
+      CallbackApiConfig callbackApiConfig,
+      OkHttpClient httpClient,
+      Gson gson) {
+    AuthHelper authHelper = buildAuthHelper(appConfig, callbackApiConfig);
     return new RestUniqueAddressIntegration(
-        sep31Config.getUniqueAddressIntegrationEndPoint(), httpClient, authHelper, gson);
+        callbackApiConfig.getBaseUrl(), httpClient, authHelper, gson);
   }
 
   @Bean
   CustomerIntegration customerIntegration(
-      Sep12Config sep12Config, OkHttpClient httpClient, AuthHelper authHelper, Gson gson) {
+      AppConfig appConfig,
+      CallbackApiConfig callbackApiConfig,
+      OkHttpClient httpClient,
+      Gson gson) {
     return new RestCustomerIntegration(
-        sep12Config.getCustomerIntegrationEndPoint(), httpClient, authHelper, gson);
+        callbackApiConfig.getBaseUrl(),
+        httpClient,
+        buildAuthHelper(appConfig, callbackApiConfig),
+        gson);
   }
 
   @Bean
   RateIntegration rateIntegration(
-      Sep38Config sep38Config, OkHttpClient httpClient, AuthHelper authHelper, Gson gson) {
+      AppConfig appConfig,
+      CallbackApiConfig callbackApiConfig,
+      OkHttpClient httpClient,
+      Gson gson) {
     return new RestRateIntegration(
-        sep38Config.getQuoteIntegrationEndPoint(), httpClient, authHelper, gson);
+        callbackApiConfig.getBaseUrl(),
+        httpClient,
+        buildAuthHelper(appConfig, callbackApiConfig),
+        gson);
   }
 
   @Bean
   FeeIntegration feeIntegration(
-      Sep31Config sep31Config, OkHttpClient httpClient, AuthHelper authHelper, Gson gson) {
+      AppConfig appConfig,
+      CallbackApiConfig callbackApiConfig,
+      OkHttpClient httpClient,
+      Gson gson) {
     return new RestFeeIntegration(
-        sep31Config.getFeeIntegrationEndPoint(), httpClient, authHelper, gson);
+        callbackApiConfig.getBaseUrl(),
+        httpClient,
+        buildAuthHelper(appConfig, callbackApiConfig),
+        gson);
+  }
+
+  AuthHelper buildAuthHelper(AppConfig appConfig, CallbackApiConfig callbackApiConfig) {
+    String authSecret = callbackApiConfig.getAuth().getSecret();
+    switch (callbackApiConfig.getAuth().getType()) {
+      case JWT_TOKEN:
+        return AuthHelper.forJwtToken(
+            new JwtService(authSecret),
+            Long.parseLong(callbackApiConfig.getAuth().getExpirationMilliseconds()),
+            appConfig.getHostUrl());
+      case API_KEY:
+        return AuthHelper.forApiKey(authSecret);
+      default:
+        return AuthHelper.forNone();
+    }
```

Comment:
❓ Is this perhaps something that can be shared with the reference server? Or maybe not worth it?

Reply:
Yes. Reference server is outside the scope of this PR. #574 is created to track this issue.